include/iceflow/dagParser.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef DAGPARSER_HPP
+#define DAGPARSER_HPP
+
+#include <fstream>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <optional>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+using json = nlohmann::json;
+
+namespace iceflow {
+
+// Structs for resources, container, scaling parameters, and communication task
+struct Resources {
+  uint32_t cpu;
+  uint32_t memory;
+};
+
+struct Container {
+  std::string image;
+  Resources resources;
+};
+
+struct ScalingParameters {
+  uint32_t taskComplexity;
+};
+
+struct CommunicationTask {
+  std::optional<uint32_t> payloadSize;
+};
+
+struct Edge {
+  std::string id;
+  std::string target;
+  uint32_t max_partitions;
+};
+
+// Paramters of a node in the DAG
```

Comment:
Small typo; could probably also be turned into a Doxygen comment.

```suggestion
/**
  * Parameters of a node in the DAG.
  */
```